File: cpp/151-160/Find Minimum in Rotated Sorted Array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int i = 0;
        int j = nums.size() - 1;
        
        while (i < j) {
            int mid = (i + j) / 2;
            if (nums[mid] < nums[j]) {
                j = mid;
            } else {
                i = mid + 1;
            }
        }
        
        return nums[i];
    }
};

File: python/151-160/Find Minimum in Rotated Sorted Array.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def findMin(self, nums):
        start = 0
        end = len(nums)-1
        
        while start < end:
            if nums[start] < nums[end]:
                return nums[start]
            
            mid = (start+end)/2
            if(nums[mid] >= nums[start]):
                start = mid+1
            else:
                end = mid

    return nums[start]